app/controllers/fhir/r4/api_controller.rb

```diff
@@ -266,6 +274,14 @@ def transaction
     contents.entry&.each_with_index do |entry, index|
       next unless entry.resource&.resourceType&.downcase == 'patient'
 
+      # Check for duplicates
+      if entry&.request&.ifNoneExist.present?
+        search_params = entry.request.ifNoneExist.split('&').map { |p| p.split('=', 2) }.select { |p| p.size == 2 }.to_h
+        num_matches = search_patients(search_params).size
+        err_msg = "There #{num_matches > 1 ? 'are' : 'is'} #{num_matches} potential duplicate patient#{num_matches > 1 ? 's' : ''}"
+        status_precondition_failed_with_custom_errors([err_msg]) && return if num_matches.positive?
```

Comment:
Could this block be a function just passing the different query strings in? So we can use it in both places.

Reply:
Also when we throw the error on 282, let's include the path to the relevant resource (should be able to get it with `Bundle.entry[#{index}].resource`. This should be placed on the `OperationOutcome.issue.expression` element.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1098/commits/b335e9bae20603fd40a4b1f465c206aff9f4b9ef#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R957

Reply:
@ngfreiter as for your first comment, I tried that and quickly realized that it might be tricky because this logic needs to return from the controller method if duplicates are detected. I cleaned up the block of code though and it's not too long so I think it's fine